packages/@tinacms/toolkit/src/components/media/media-manager.tsx

```diff
@@ -197,6 +264,53 @@ export function MediaPicker({
     },
   })
 
+  const syncMedia = async () => {
+    if (hasTinaMedia) {
+      const res = await cms.api.tina.syncTinaMedia()
+      if (res?.assetsSyncing) {
+        // it was successful
+        try {
+          setListState('loading')
+          // poll every 3 seconds until it is done
+          await poll(
+            async () => {
+              const status = await cms.api.tina.checkSyncStatus({
+                assetsSyncing: res.assetsSyncing,
+              })
+              const totalDone = Object.values(status.status).filter(
+                Boolean
+              )?.length
+              const total = Object.keys(status.status)?.length
+              setLoadingText(`${totalDone}/${total} Media items loaded`)
+              return status
+            },
+            3000,
+            // Will time out after 10 seconds
+            10000
```

Comment:
might want this to be higher than 10 seconds. we can test & tweak later, but when I was testing w/ a large amount of big-ish files, it definitely took more than 10 seconds.

Reply:
Sounds good!